common/src/abstractions/state.service.ts

```diff
@@ -265,6 +265,10 @@ export abstract class StateService<T extends Account = Account> {
   ) => Promise<void>;
   getPasswordGenerationOptions: (options?: StorageOptions) => Promise<any>;
   setPasswordGenerationOptions: (value: any, options?: StorageOptions) => Promise<void>;
+  getUsernameGenerationOptions: (options?: StorageOptions) => Promise<any>;
```

Comment:
Could we add a proper return type for this?

Reply:
We could, but I feel like we need to do more refactoring work for all generation options that currently exist as well to keep things consistent.  Maybe a tech debt task?

Reply:
I think we could quite easily add an interface for the option, and just use `Partial<UsernameGeneratorOptions>` for all the functions that expects it. And leave password as a tech debt.

But I leave that up to you to decide.